### PR TITLE
Metal3 i18n form

### PR DESCRIPTION
**Pull request by yaacov**
More Metal3 plugin i18n strings.

Missing from this PR - Actions, Menu Items and Tabs from `plugin.json` file.

screenshot-localhost_9000-2020 11 25-20_58_07
screenshot-localhost_9000-2020 11 25-20_59_47
screenshot-localhost_9000-2020 11 25-21_00_46
screenshot-localhost_9000-2020 11 25-22_24_09

### CODE REVIEW

**Comment by yaacov** [comment]
@pcbailey @rawagner @glekner @metalice please review

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/add-baremetal-host/AddBareMetalHost.tsx [hunk 1]
```diff
@@ -131,20 +133,27 @@ const AddBareMetalHost: React.FC<AddBareMetalHostProps> = ({
       name: Yup.mixed()
         .test(
           'unique-name',
-          'Name "${value}" is already taken.', // eslint-disable-line no-template-curly-in-string
+          t('metal3-plugin~Name ${value} is already taken.'), // eslint-disable-line no-template-curly-in-string
```
Comment (yaacov) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/add-baremetal-host/AddBareMetalHost.tsx:136]:
```suggestion
          t('metal3-plugin~Name "${value}" is already taken.'), // eslint-disable-line no-template-curly-in-string
```

File: frontend/packages/metal3-plugin/locales/en/metal3-plugin.json [hunk 2]
```diff
@@ -59,6 +83,85 @@
...
+  "Restart": "Restart",
+  "Start Maintenance": "Start Maintenance",
+  "All managed workloads will be moved off of this node. New workloads and data will not be added to this node until maintenance is stopped.": "All managed workloads will be moved off of this node. New workloads and data will not be added to this node until maintenance is stopped.",
+  "If the node does not exit maintenance within <1>30 minutes</1>, the cluster will automatically rebuild the node&apos;s data using replicated copies": "If the node does not exit maintenance within <1>30 minutes</1>, the cluster will automatically rebuild the node&apos;s data using replicated copies",
+  "The Ceph storage cluster is not in a healthy state.": "The Ceph storage cluster is not in a healthy state.",
+  "Maintenance should not be started until the health of the storage cluster is restored.": "Maintenance should not be started until the health of the storage cluster is restored.",
+  "Are you sure you want to stop maintenance <1>{reason ? ` (${reason}) ` : ''}</1>on node <3>{nodeName}</3>?": "Are you sure you want to stop maintenance <1>{reason ? ` (${reason}) ` : ''}</1>on node <3>{nodeName}</3>?",
```
Comment (yaacov) [inline frontend/packages/metal3-plugin/locales/en/metal3-plugin.json:164]:
```suggestion
  "Are you sure you want to stop maintenance <1>{reason ? ` (${reason}) ` : ''}</1>on node <3>{nodeName}</3>?": "Are you sure you want to stop maintenance <1>{reason}</1>on node <3>{nodeName}</3>?",
```

**Review by jtomasek — APPROVED** [review]

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx [hunk 3]
```diff
@@ -20,56 +21,60 @@ import BareMetalNodeStatus from '../BareMetalNodeStatus';
...
 
 const StatusCard: React.FC = () => {
+  const { t } = useTranslation();
   const { obj } = React.useContext(NodeDashboardContext);
   const { nodeMaintenance, csr, host } = React.useContext(BareMetalNodeDashboardContext);
   const bmoEnabled = useFlag(BMO_ENABLED_FLAG);
   const status = bareMetalNodeStatus({ node: obj, nodeMaintenance, csr });
   const hasPowerMgmt = hasPowerManagement(host);
+
+  const POWER_MGMT_MSG = t(
```
Comment (rawagner) [inline frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx:33]: can you inline this ?

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx [hunk 4]
```diff
@@ -20,56 +21,60 @@ import BareMetalNodeStatus from '../BareMetalNodeStatus';
...
+    'metal3-plugin~Power operations cannot be performed on this host until Baseboard Management Controller (BMC) credentials are provided for the underlying host.',
+  );
+
+  const POWER_MGMT_ALERT = () => ({
```
Comment (rawagner) [inline frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx:37]: why is this a function ? 
Can you also use camelcase ?

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx [hunk 5]
```diff
@@ -20,56 +21,60 @@ import BareMetalNodeStatus from '../BareMetalNodeStatus';
...
+            )}/edit?powerMgmt`}
+          >
+            {t('metal3-plugin~Add credentials')}
+          </Link>
+        </StackItem>
+      </Stack>
+    ),
+  });
+
+  const BMO_DISABLED_ALERT = {
```
Comment (rawagner) [inline frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx:57]: i'd use camelcase here too

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx (hunk 4 above)
Comment (rawagner) [reply frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx:37]: Can you also inline it together with BMO_DISABLED_ALERT ?

File: frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx [hunk 6]
```diff
@@ -28,36 +30,48 @@ import { StatusProps } from '../types';
...
-);
+const SafePowerOffDialog: React.FC<SafePowerOffDialogProps> = ({ isUnderMaintenance, pods }) => {
+  const { t } = useTranslation();
+  return (
+    <p>
+      {t('metal3-plugin~Host is ready to be gracefully powered off.')}{' '}
+      {isUnderMaintenance && (
+        <p>
+          {t('metal3-plugin~The host is currently under maintenance and {{powerOffMessage}}', {
+            powerOffMessage: getPowerOffMessage(t, pods),
```
Comment (rawagner) [inline frontend/packages/metal3-plugin/src/components/modals/PowerOffHostModal.tsx:68]: I think we should not construct sentences anymore

**Review by rawagner — CHANGES_REQUESTED** [review]

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/table-filters.ts [hunk 7]
```diff
@@ -1,23 +1,24 @@
 import * as _ from 'lodash';
+import { TFunction } from 'i18next';
 import { RowFilter } from '@console/internal/components/filter-toolbar';
 import { NODE_STATUS_TITLE_KEYS } from '../../constants';
 import { BareMetalNodeListBundle, isCSRBundle } from '../types';
 
 const statesToFilterMap = Object.freeze({
   ready: {
-    title: 'Ready',
+    titleKey: 'Ready',
```
[the remark this answers is not in the document]
Comment (yaacov) [reply frontend/packages/metal3-plugin/src/components/baremetal-nodes/table-filters.ts:9]: wow - thanks !

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/CSRStatus.tsx [hunk 8]
```diff
@@ -34,19 +30,28 @@ const CSRStatus: React.FC<CSRStatusProps> = ({ csr, title, serverCSR }) => {
       setInProgress(false);
     }
   };
+
+  const CLIENT_CSR_DESC = t(
```
Comment (yaacov) [inline frontend/packages/metal3-plugin/src/components/baremetal-nodes/CSRStatus.tsx:34]: make cammelcase

File: frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx [hunk 9]
```diff
@@ -21,55 +22,62 @@ import { BMO_ENABLED_FLAG } from '../../../features';
 import { hasPowerManagement } from '../../../selectors';
 import { BareMetalHostModel } from '../../../models';
 import { BareMetalHostKind } from '../../../types';
+import { TFunction } from 'i18next';
```
Comment (yaacov) [inline frontend/packages/metal3-plugin/src/components/baremetal-nodes/dashboard/StatusCard.tsx:25]: move to top of file

**Pull request merged**